packages/core/content-type-builder/admin/src/utils/parseDateValue.ts

```diff
@@ -0,0 +1,18 @@
+import { Schema } from '@strapi/types';
+
+export const parseDateValue = <TAttribute extends Schema.Attribute.AnyAttribute>(
+  value: Schema.Attribute.Value<TAttribute>
+): Date | undefined => {
```

Comment:
you don't really need to specify the return type here, typescript will infer it correctly